### PR TITLE
P4RT-5.2  deviation for handling ttl0 and hoplimit0 packets

### PR DESCRIPTION
**Pull request by singh-prem**
1. Added a deviation for handling ttl=0 and hoplimit=0 packets.
In ideal conditions, ttl=0 and hoplimit=0 packets are illegal packets on the network and should not be forwarded by an upstream device.
2. minor fix in p4rtutils.go `inferP4RTNodesCisco`

### CODE REVIEW

**Comment by singh-prem** [comment]
logs:
`
[linux]$ go test -timeout 0 -args -deviation_explicit_p4rt_node_component=true -deviation_p4rt_ttl0_hoplimit0=true -testbed atedut.testbed -binding atedut.binding -v 5

*** Reserving the testbed...

...

*** PROPERTY: topology -> ate:8,dut:8

********************************************************************************

  Testbed Reservation Complete
  ID: STATIC

    dut:              DUT
    port6:            FourHundredGigE0/0/0/15
    port7:            FourHundredGigE0/0/0/16
    port8:            FourHundredGigE0/0/0/17
    port1:            FourHundredGigE0/0/0/10
    port2:            FourHundredGigE0/0/0/11
    port3:            FourHundredGigE0/0/0/12
    port4:            FourHundredGigE0/0/0/13
    port5:            FourHundredGigE0/0/0/14
    ate:              20.0.0.2
    port8:            1/8
    port1:            1/1
    port2:            1/2
    port3:            1/3
    port4:            1/4
    port5:            1/5
    port6:            1/6
    port7:            1/7

********************************************************************************

=== RUN   TestPacketOut
    traceroute_packetout_test.go:236: 
        *** Pushing topology to 20.0.0.2...
        
        
    traceroute_packetout_test.go:236: 
        *** Starting protocols on 20.0.0.2...
        
        
    traceroute_packetout_test.go:148: Configuring P4RT Node: 0/RP0/CPU0-NPU0
    traceroute_packetout_test.go:243: 
        *** Fetching P4RT client for DUT...
        
        
=== RUN   TestPacketOut/PacketOut_from_Primary_Controller_ipv4_
    packetout_test.go:76: Sending ipv4 pakcets with ttl 1
    packetout_test.go:78: Initial number of packets: 1
    packetout_test.go:82: Sending packets now
    packetout_test.go:91: Final number of packets: 106
    packetout_test.go:94: Received 105 packets on ATE port 1/1
=== RUN   TestPacketOut/PacketOut_from_Primary_Controller_ipv6
    packetout_test.go:116: Sending ipv6 packets with ttl = 1
    packetout_test.go:118: Initial number of packets: 106
    packetout_test.go:122: Sending packets now
    packetout_test.go:131: Final number of packets: 206
    packetout_test.go:134: Received 100 packets on ATE port 1/1
--- PASS: TestPacketOut (255.77s)
    --- PASS: TestPacketOut/PacketOut_from_Primary_Controller_ipv4_ (80.06s)
    --- PASS: TestPacketOut/PacketOut_from_Primary_Controller_ipv6 (80.04s)
PASS

*** Releasing the testbed...

*** PROPERTY: time.begin -> 1675059658
*** PROPERTY: time.end -> 1675059927
ok  	github.com/openconfig/featureprofiles/feature/experimental/p4rt/ate_tests/traceroute_packetout_test	269.318s
`

**Review by thesrinath — DISMISSED** [review]

File: feature/experimental/p4rt/ate_tests/traceroute_packetout_test/packetout_test.go [hunk 1]
```diff
@@ -63,6 +64,9 @@ func testPacketOut(ctx context.Context, t *testing.T, args *testArgs) {
 	leader := args.leader
 	desc := "PacketOut from Primary Controller"
 	ttls := []int{0, 1}
+	if *deviations.P4rtTTLHoplimitZero {
```
Comment (vosipchu) [inline feature/experimental/p4rt/ate_tests/traceroute_packetout_test/packetout_test.go:67]: Why do we have a deviation here? As of the latest HLD, we only care about TTL=1. No support for 0 is normal.
Can we please update this PR?

Comment (vbhan-cisco) [reply feature/experimental/p4rt/ate_tests/traceroute_packetout_test/packetout_test.go:67]: Sure we will remove TTL0 from this test

**Comment by xw-g** [comment]
I think LGTM from @vosipchu and @Ankur19 should be good enough. Internal b/267356082 for presubmit effort.

File: feature/experimental/p4rt/ate_tests/traceroute_packetout_test/packetout_test.go [hunk 2]
```diff
@@ -62,86 +62,63 @@ func sendPackets(t *testing.T, client *p4rt_client.P4RTClient, packets []*p4v1.P
...
-				if counter1-counter0 > uint64(float64(packetCounter)*0.05) {
-					t.Fatalf("Ttl=0 packets are being forwarded.")
-				}
-			} else {
-				if counter1-counter0 < uint64(float64(packetCounter)*0.95) {
-					t.Fatalf("Not all the packets are received.")
-				}
-			}
-			time.Sleep(20 * time.Second)
+		if counter1-counter0 < uint64(float64(packetCounter)*0.95) {
```
Comment (Ankur19) [inline feature/experimental/p4rt/ate_tests/traceroute_packetout_test/packetout_test.go:90]: nit: Can you lift the `0.95` up to the top of the file and declare it as a variable say `allowCounterDiff = 0.95` and use it in all the places. This helps in debugging later.

**Review by Ankur19 — APPROVED** [review]
LGTM!

**Review by vosipchu — APPROVED** [review]

**Review by thesrinath — APPROVED** [review]

**Pull request merged**